app/controllers/common/FileUploadHelper.scala

```diff
@@ -85,6 +86,12 @@ case class FileUploadHelper @Inject() (
             }
         case file: UploadedFile.Success   =>
           removeFile(mode, draftId, file.fileUrl.get, isLetterOfAuthority)
+        case _: Failure                   =>
+          Future.failed(
+            new RuntimeException(
+              "Shouldn't reach this point. Failure state only generated by Duplicate filenames which is handled elsewhere"
+            )
+          )
```

Comment:
This one's a bit annoying. The case type UploadedFile comtains Initiated, Success and Failure. But failure is only set after coming back to the Upload page and is only ever used to reject duplicate filenames. That happens after this point.

Reply:
I'm not sure just throwing an exception works best here - an exception could use a more precise error message or we could rework this to return a Left with some logging of useful details.

Plus we don't know if the libraries behaviour around that might change later on.